### DISCUSSION TITLE
Adapt this for "remote" monitoring

### DISCUSSION THREAD

**Discussion by newhinton** [klaasnicolaas/home-assistant-glow#278] [General]
I dont have wifi access at my meter to make use of the project as-is. However, i can use a neighbours wifi to access the internet, and therefore my home-assistant instance.

However, i have quite a hard time to get the power-meter-data into homeassistant, especially with history and all the features home assistant provides.

One idea i had was to use a secondary esp, to act as a virtual-power-meter, so that the actual remote device can update the virtual one through home assistant's services. That does not work too reliable. Does anyone have an idea on how to make this work? Thanks!

**Comment by bigwave** [comment]
You can do this using the [Wireguard component ](https://esphome.io/components/wireguard.html)

It's not trivial to set up but once configured correctly it's robust and easy to add other remote devices.
1. Install Wireguard server on the same network as Home Assistant.
2. Configure it and the firewall (you can test using a mobile phone on mobile data)
3. Use an ESP32 for the remote monitoring device (ESP8266 isn't powerful enough for Wireguard)
4. Install and configure glow
5. Adopt into ESPHome addon
6. Add Wireguard configuration through ESPHome addon (you can test using wifi details of a mobile phone wifi hotspot using mobile data)
7. Add remote location wifi details to config
8. Deploy at remote location
